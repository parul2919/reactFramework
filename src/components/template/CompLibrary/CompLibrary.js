import React from "react";
import styled from "styled-components";
import styles from "./CompLibrary.style";
import MemoizedFooter from "../../organism/Footer/Footer";
import MemoizedHeader from "../../organism/Header/Header";
import ComponentGuide from "../../organism/ComponentGuide/ComponentGuide";

const DummyTemplate2 = (props) => {
  const { className } = props;
  return (
    <div className={className}>
      <MemoizedHeader />
      <ComponentGuide />
      <MemoizedFooter />
    </div>
  );
};

export default styled(DummyTemplate2)`
  ${styles}
`;
export { DummyTemplate2 };
