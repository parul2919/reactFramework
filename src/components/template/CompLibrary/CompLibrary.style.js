import { css } from "styled-components";

export default css`
  &.dummy-template {
    color: black;
  }
`;
