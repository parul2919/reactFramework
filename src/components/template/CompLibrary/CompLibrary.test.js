import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import CompLibrary from "./CompLibrary";

describe("<Component /> spec", () => {
  it("renders the component", () => {
    const view = render(
      <Router>
        <CompLibrary />
      </Router>
    );
    expect(view).toMatchSnapshot();
  });
});
