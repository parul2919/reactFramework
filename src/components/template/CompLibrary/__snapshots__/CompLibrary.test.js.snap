// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Component /> spec renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hKMtZM gPhcHH"
      >
        <header
          class="header sc-gsnTZi jOpTsK"
        >
          <nav
            class="navbar navbar-expand d-flex align-items-center justify-content-between"
          >
            <a
              class="navbar-brand "
              href="/"
            >
              <img
                alt="logo-img"
                src="test-file-stub"
                width="150"
              />
            </a>
            <div
              class="collapse navbar-collapse"
            >
              <ul
                class="navbar-nav ml-auto "
              >
                <a
                  class="nav-link"
                  href="/"
                  style="text-decoration: underline;"
                  target="_self"
                >
                  Tech Stacks
                </a>
                <a
                  class="nav-link"
                  href="/component-library"
                  style="text-decoration: none;"
                  target="_self"
                >
                  Component Library
                </a>
                <a
                  class="nav-link"
                  href="/project-structure"
                  style="text-decoration: none;"
                  target="_self"
                >
                  Project Structure
                </a>
              </ul>
            </div>
          </nav>
        </header>
        <section
          class="techStack wrapper-std sc-dkzDqf ivVfSh"
        >
          <div
            class="techStack-list"
          >
            <h2>
              Do find the tech-stack list which we are using to build this architecure
            </h2>
            <h3>
              1. Component Structure Definition
            </h3>
            <ul>
              <li>
                Atom
              </li>
              <li>
                Molecule
              </li>
              <li>
                Organism
              </li>
              <li>
                Template
              </li>
              <li>
                Page
              </li>
            </ul>
            <h3>
              2. Defined Atom Components
            </h3>
            <ul>
              <li>
                ButtonAtom
              </li>
              <li>
                DownloadCSV
              </li>
              <li>
                ErrorBoundry
              </li>
              <li>
                FlashMessage - 
              </li>
              <li>
                Loader - Loader component to be used throughout application
              </li>
              <li>
                Select - selectbox where only one option can be selected
              </li>
              <li>
                Multiselect - selectbox where multiple options can be selected
              </li>
              <li>
                Pagination
              </li>
              <li>
                TabButton
              </li>
              <li>
                ToggleSwitch
              </li>
              <li>
                Tooltip
              </li>
            </ul>
            <h3>
              3. Defined Molecule Components
            </h3>
            <ul>
              <li>
                Overlay Modal
              </li>
              <li>
                Accordion
              </li>
              <li>
                Confirm Popover
              </li>
              <li>
                Date Range Selector
              </li>
              <li>
                Tabs
              </li>
            </ul>
            <h3>
              4. Defined Organism Components
            </h3>
            <ul>
              <li>
                Header
              </li>
              <li>
                AboutForm
              </li>
              <li>
                Footer
              </li>
            </ul>
            <h3>
              5. Helper Modules and Libraries Implementation 
            </h3>
            <ul>
              <li>
                moment js Implementation ex: 
              </li>
              <li>
                 env-cmd Implementation
              </li>
              <li>
                csv export Implementation
              </li>
            </ul>
          </div>
        </section>
        <section
          class="footer sc-bczRLJ eRSPWO"
        >
          <nav
            style="max-width: 1440px; margin: auto;"
          >
            <a
              class="footer-logo"
              href="/"
            >
              <img
                alt="logo-img"
                src="test-file-stub"
                width="150"
              />
            </a>
          </nav>
          <div
            class="disclaimer-text text-center"
          >
            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ips um has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hKMtZM gPhcHH"
    >
      <header
        class="header sc-gsnTZi jOpTsK"
      >
        <nav
          class="navbar navbar-expand d-flex align-items-center justify-content-between"
        >
          <a
            class="navbar-brand "
            href="/"
          >
            <img
              alt="logo-img"
              src="test-file-stub"
              width="150"
            />
          </a>
          <div
            class="collapse navbar-collapse"
          >
            <ul
              class="navbar-nav ml-auto "
            >
              <a
                class="nav-link"
                href="/"
                style="text-decoration: underline;"
                target="_self"
              >
                Tech Stacks
              </a>
              <a
                class="nav-link"
                href="/component-library"
                style="text-decoration: none;"
                target="_self"
              >
                Component Library
              </a>
              <a
                class="nav-link"
                href="/project-structure"
                style="text-decoration: none;"
                target="_self"
              >
                Project Structure
              </a>
            </ul>
          </div>
        </nav>
      </header>
      <section
        class="techStack wrapper-std sc-dkzDqf ivVfSh"
      >
        <div
          class="techStack-list"
        >
          <h2>
            Do find the tech-stack list which we are using to build this architecure
          </h2>
          <h3>
            1. Component Structure Definition
          </h3>
          <ul>
            <li>
              Atom
            </li>
            <li>
              Molecule
            </li>
            <li>
              Organism
            </li>
            <li>
              Template
            </li>
            <li>
              Page
            </li>
          </ul>
          <h3>
            2. Defined Atom Components
          </h3>
          <ul>
            <li>
              ButtonAtom
            </li>
            <li>
              DownloadCSV
            </li>
            <li>
              ErrorBoundry
            </li>
            <li>
              FlashMessage - 
            </li>
            <li>
              Loader - Loader component to be used throughout application
            </li>
            <li>
              Select - selectbox where only one option can be selected
            </li>
            <li>
              Multiselect - selectbox where multiple options can be selected
            </li>
            <li>
              Pagination
            </li>
            <li>
              TabButton
            </li>
            <li>
              ToggleSwitch
            </li>
            <li>
              Tooltip
            </li>
          </ul>
          <h3>
            3. Defined Molecule Components
          </h3>
          <ul>
            <li>
              Overlay Modal
            </li>
            <li>
              Accordion
            </li>
            <li>
              Confirm Popover
            </li>
            <li>
              Date Range Selector
            </li>
            <li>
              Tabs
            </li>
          </ul>
          <h3>
            4. Defined Organism Components
          </h3>
          <ul>
            <li>
              Header
            </li>
            <li>
              AboutForm
            </li>
            <li>
              Footer
            </li>
          </ul>
          <h3>
            5. Helper Modules and Libraries Implementation 
          </h3>
          <ul>
            <li>
              moment js Implementation ex: 
            </li>
            <li>
               env-cmd Implementation
            </li>
            <li>
              csv export Implementation
            </li>
          </ul>
        </div>
      </section>
      <section
        class="footer sc-bczRLJ eRSPWO"
      >
        <nav
          style="max-width: 1440px; margin: auto;"
        >
          <a
            class="footer-logo"
            href="/"
          >
            <img
              alt="logo-img"
              src="test-file-stub"
              width="150"
            />
          </a>
        </nav>
        <div
          class="disclaimer-text text-center"
        >
          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ips um has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
